docs/grid/column-filtering.md

```diff
@@ -0,0 +1,355 @@
+# Column Filtering
+
+Column filtering in Highcharts Grid allows users to filter data based on specific conditions and values for each column. This powerful feature enhances data exploration and helps users focus on relevant information within large datasets.
+
+## Basic setup
+
+To enable column filtering, you need to set the `filtering.enabled` option to `true` for the columns you want to make filterable.
+
+```js
+{
+    columnDefaults: {
+        filtering: {
+            enabled: true
+        }
+    },
+    columns: [{
+        id: "product",
+        filtering: {
+            enabled: true
+        }
+    }, {
+        id: "price",
+        filtering: {
+            enabled: true
+        }
+    }]
+}
+```
+
+## Filtering Interface
+
+Column filtering supports two different interface modes:
+
+### Inline Filtering
+
+When `inline` is set to `true`, filter inputs are rendered directly in a special header row below the column headers.
+
+```js
+{
+    columnDefaults: {
+        filtering: {
+            enabled: true,
+            inline: true
+        }
+    }
+}
+```
+
+### Popup Filtering
+
+When `inline` is set to `false` (default), filtering is accessed via a popup that opens when clicking the filter icon in the column header.
```

Comment:
Here, I'd rephrase it to indicate that when popup filtering, the filter icon is rendered, and then it triggers the popup.